torchinductor/lowering.py

```diff
@@ -88,6 +89,23 @@ def decode_dtype(dtype: int):
     return dtype
 
 
+def is_integer_type(x):
+    if isinstance(x, TensorBox):
+        return x.get_dtype() in {
```

Comment:
perhaps 
```
return not x.get_dtype().is_floating_point
```

Reply:
this won't be future proof, as we'd add `complex` types at some point. But perhaps you could use `torch._prims_common.is_integer_dtype`.